tests/CopyFilesTaskIntegrationTest.php

```diff
@@ -13,24 +14,28 @@ class CopyFilesTaskIntegrationTest extends \PHPUnit\Framework\TestCase
 
 	private const TEMP_DIRECTORY_PATH = __DIR__ . '/temp';
 
-	public function testCopyFile(): void
+	/**
+	 * @return mixed[][]|\Generator
+	 */
+	public function copyFileDataProvider(): Generator
```

Comment:
Now the fact that the two cases were same was more of an accident, than design choice and so like this the data provider does not make much sense here. Plus I would argue it also decreases readability by a significant degree.

So at first I just thought not to use them here, but upon analysing the other cases here I think we can have a DP for "single file copy", but then all of the relevant things must be as parameters:

* target
* source and target file (relative to the temp/ dir)
* existing source and (optionally) target content
* expected target content
* expected log message

I think like this the relevant parts for each case will be in DP and "boilerplate" will be in test implementation.

Reply:
Fair point, more test cases can be deduplicated using this approach.